backend/util/FileUploadController.js

```diff
@@ -0,0 +1,158 @@
+// Code adapted from youtube - @ Sam Meech-Ward
+// @youtube https://www.youtube.com/watch?v=NZElg91l_ms
+// @github https://github.com/Sam-Meech-Ward/image-upload-s3
+// @author Mohak Vaswani
+
+const fs = require("fs");
+const util = require("util");
+const File = require("../models/File_model");
+
+const unlinkFile = util.promisify(fs.unlink);
+
+const { uploadFile, getFileStream, DeleteFile } = require("./S3Util");
+
+const Upload_File = async (req, res) => {
+  try {
+    const file = req.file;
+    const user = req.user;
+    const result = await uploadFile(file);
+    await unlinkFile(file.path);
+    await File.create({
+      name: req.body.name,
+      userID: user._id,
+      Category_ID: "test",
+      S3_ID: result.key,
+    });
+    console.log("File uploaded successfully");
+    res.json({ data: "success!" });
+    /*
+    res.redirect("/file/?secret_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyIjp7Il9pZCI6IjYxZmEyMDQwMWMxOThlOTY3Y2IwYWU5ZSIsImVtYWlsIjoibW9oYWt2MTVAZ21haWwuY29tIn0sImlhdCI6MTY0MzgyNzAwM30.a6oz0g3miTQMUpSSU-2MMkpBVi2-lZFhsz_pfQg4X_U"); */
+  } catch (e) {
+    console.log(e);
+    res.status(400).json({ error: "Error, cannot upload file" });
+  }
+};
+
+const Delete_File = async (req, res) => {
+  try {
+    const ID = req.params.s3id;
+    File.findOne({ S3_ID: ID })
```

Comment:
We should be finding based on ID (https://mongoosejs.com/docs/guide.html#id) not based on S3_id. The amazon s3 id should only ever be seen by our backend. The frontend and backend should interface with _id generated by mongoose.

Reply:
This goes for every time we are searching by id